packages/training/python/src/training/hybrid_env.py

```
@@ -0,0 +1,522 @@
+        # If offline selected but no trajectories, use online
+        if not use_online and len(self.trajectory_cache) == 0:
+            use_online = True
+        
+        if use_online:
+            self.online_count += 1
+            return await self._get_online_item()
+        else:
+            self.offline_count += 1
+            return self._get_offline_item()
+    
+    async def _get_online_item(self) -> Tuple["PoolScenario", str]:
```

Ruff (0.14.10)

Undefined name `PoolScenario`

(F821)